src/test/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorderTest.java

```diff
@@ -0,0 +1,104 @@
+package com.aspectsecurity.contrast.contrastjenkins;
+
+import com.contrastsecurity.http.TraceFilterForm;
+import com.contrastsecurity.models.Application;
+import com.contrastsecurity.models.Applications;
+import com.contrastsecurity.models.Traces;
+import com.contrastsecurity.sdk.ContrastSDK;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildListener;
+import hudson.model.Result;
+import hudson.model.TaskListener;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.spy;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Jenkins.class, VulnerabilityTrendRecorder.class, VulnerabilityTrendHelper.class})
+public class VulnerabilityTrendRecorderTest {
+
+    @Mock
+    Jenkins jenkins;
+    @Mock
+    VulnerabilityTrendRecorder.DescriptorImpl vulnerabilityTrendRecorderDescriptor;
+
+
+    @Before
+    public void setUp() {
+        PowerMockito.mockStatic(Jenkins.class);
+        PowerMockito.mockStatic(VulnerabilityTrendRecorder.class);
+        PowerMockito.mockStatic(VulnerabilityTrendHelper.class);
```

Comment:
Is it possible to not mock any of these classes? Avoid mocks if at all possible.

Reply:
Mocks are required here to setup the Jenkins environment.